### PR TITLE
[JENKINS-59857] Kerberos support updated

### PR DESCRIPTION
**Pull request by Emil-Gustafsson**
see [JENKINS-59857](https://issues.jenkins-ci.org/browse/JENKINS-59857)

based on Vice Radman Kerberos support pull request.
Fixed exception handling.

### CODE REVIEW

**Comment by kuisathaverat** [comment]
I've found a GitHub project that could help to test this https://github.com/criteo/kerberos-docker

**Comment by Emil-Gustafsson** [comment]
this looks like a big change to add, i think it should be its own pull request

**Comment by Arts13** [comment]
Hello how can i can get ssh host message?

File: src/com/trilead/ssh2/Connection.java [hunk 1]
```diff
@@ -341,6 +341,29 @@ public synchronized boolean authenticateWithPassword(String user, String passwor
 
 		return authenticated;
 	}
+	
+	public synchronized boolean authenticateWithGssapiWithMic(String user, String host) throws IOException {
```
Comment (Emil-Gustafsson) [inline src/com/trilead/ssh2/Connection.java]: can "host" be this.hostname instead or is this something else?

**Comment by Emil-Gustafsson** [comment]
manually tested the implementation:

> 2019-09-06 15:04:02,121 (AbstractCli.java:80) DEBUG : Connecting cli
> 2019-09-06 15:04:02,125 (ImplSsh.java:435) DEBUG : Creating a ssh connection
> 2019-09-06 15:04:02,585 (ImplSsh.java:449) DEBUG : Done creating a ssh connection
> 2019-09-06 15:04:02,598 (ImplSsh.java:373) DEBUG : Available authentication methods: [publickey, gssapi-keyex, gssapi-with-mic, password, keyboard-interactive]
> 2019-09-06 15:04:02,598 (ImplSsh.java:375) DEBUG : Ignoring authentication methods: [publickey, password, keyboard-interactive]
> 2019-09-06 15:04:02,598 (ImplSsh.java:385) DEBUG : Trying to authenticate with method: gssapi-keyex
> 2019-09-06 15:04:02,598 (ImplSsh.java:453) DEBUG : Reconnecting ssh connection
> 2019-09-06 15:04:02,598 (ImplSsh.java:137) DEBUG : Disconnecting ssh connection
> 2019-09-06 15:04:02,599 (ImplSsh.java:153) DEBUG : Done disconnecting ssh connection
> 2019-09-06 15:04:02,600 (ImplSsh.java:435) DEBUG : Creating a ssh connection
> 2019-09-06 15:04:02,646 (ImplSsh.java:449) DEBUG : Done creating a ssh connection
> 2019-09-06 15:04:02,646 (ImplSsh.java:456) DEBUG : Done reconnecting ssh connection
> 2019-09-06 15:04:02,646 (ImplSsh.java:385) DEBUG : Trying to authenticate with method: gssapi-with-mic
> 2019-09-06 15:04:05,515 (ImplSsh.java:406) DEBUG : gssapi-with-mic authentication successful? true
> 2019-09-06 15:04:05,515 (ImplSsh.java:414) DEBUG : Authentication Complete
> 2019-09-06 15:04:05,516 (ImplSsh.java:222) DEBUG : Opening a ssh session
> 2019-09-06 15:04:05,556 (ImplSsh.java:229) DEBUG : Done opening a ssh session 
> 2019-09-06 15:04:05,557 (ImplSsh.java:315) DEBUG : Request pty: dumb with size: [WidthCharacters=0, HeightCharacters=0, WidthPixels=0, HeightPixels=0]
> 2019-09-06 15:04:05,559 (ImplSsh.java:240) DEBUG : Starting shell session
> 2019-09-06 15:04:05,561 (ImplSsh.java:246) DEBUG : Done starting shell session

**Comment by kuisathaverat** [comment]
I've made a PR to your fork with a couple of tests and a script tu run them, as soon as those changes where here we can merge and release it.

**Pull request merged**